cpp/cluster/ClusterProperties.cc

```diff
@@ -45,47 +47,50 @@ void ClusterProperties::compute(const freud::locality::NeighborQuery* nq, const
         const unsigned int c = cluster_idx[i];
         cluster_points[c].push_back((*nq)[i]);
         m_cluster_sizes[c]++;
+        m_cluster_masses[c] += masses[i];
     }
 
     // Now that we have located all of the cluster vectors, compute the centers
+
     for (unsigned int c = 0; c < num_clusters; c++)
     {
-        m_cluster_centers[c] = nq->getBox().centerOfMass(cluster_points[c].data(), m_cluster_sizes[c]);
+        m_cluster_centers[c]
+            = nq->getBox().centerOfMass(cluster_points[c].data(), m_cluster_sizes[c], masses);
     }
 
     // Now that we have determined the centers of mass for each cluster, tally
-    // up the gyration tensor. This has to be done in a loop over the points.
+    // up the moment of inertia tensor. This has to be done in a loop over the points.
     for (unsigned int i = 0; i < nq->getNPoints(); i++)
     {
         unsigned int c = cluster_idx[i];
         vec3<float> pos = (*nq)[i];
         vec3<float> delta = nq->getBox().wrap(pos - m_cluster_centers[c]);
 
         // get the start pointer for our 3x3 matrix
-        m_cluster_gyrations(c, 0, 0) += delta.x * delta.x;
-        m_cluster_gyrations(c, 0, 1) += delta.x * delta.y;
-        m_cluster_gyrations(c, 0, 2) += delta.x * delta.z;
-        m_cluster_gyrations(c, 1, 0) += delta.y * delta.x;
-        m_cluster_gyrations(c, 1, 1) += delta.y * delta.y;
-        m_cluster_gyrations(c, 1, 2) += delta.y * delta.z;
-        m_cluster_gyrations(c, 2, 0) += delta.z * delta.x;
-        m_cluster_gyrations(c, 2, 1) += delta.z * delta.y;
-        m_cluster_gyrations(c, 2, 2) += delta.z * delta.z;
+        m_cluster_inertia_moments(c, 0, 0) += delta.x * delta.x * masses[i];
+        m_cluster_inertia_moments(c, 0, 1) += delta.x * delta.y * masses[i];
+        m_cluster_inertia_moments(c, 0, 2) += delta.x * delta.z * masses[i];
+        m_cluster_inertia_moments(c, 1, 0) += delta.y * delta.x * masses[i];
+        m_cluster_inertia_moments(c, 1, 1) += delta.y * delta.y * masses[i];
+        m_cluster_inertia_moments(c, 1, 2) += delta.y * delta.z * masses[i];
+        m_cluster_inertia_moments(c, 2, 0) += delta.z * delta.x * masses[i];
+        m_cluster_inertia_moments(c, 2, 1) += delta.z * delta.y * masses[i];
+        m_cluster_inertia_moments(c, 2, 2) += delta.z * delta.z * masses[i];
     }
 
-    // Normalize by the cluster size.
+    // Normalize by the cluster masses.
```

Comment:
As discussed in the issue, we should not normalize the moment of inertia tensor by cluster masses. Only the gyration tensor is normalized by the number of particles in the cluster. For this reason, the formulae of gyration tensor and moment of inertia tensor differ by normalization as well as the consideration of per-particle mass. Because of that (and to avoid API breakage), I would recommend computing both properties (gyration tensor and moment of inertia tensor properties).